include/otf2xx/reference_generator.hpp

```diff
@@ -81,7 +81,7 @@ class reference_generator
     template <typename Definition>
     void register_definition(const Definition& def)
     {
-        static_assert(std::is_convertible<otf2::reference<Definition>, RefType>::value,
+        static_assert(std::is_convertible<typename Definition::reference_type, RefType>::value,
```

Comment:
Once the references only have a tag, this can be changed to `is_same`. `is_convertible` is way more expensive